incubator/kafka/templates/statefulset.yaml

```diff
@@ -11,6 +11,10 @@ spec:
     {{- include "kafka.broker.matchLabels" . | nindent 6 }}
   serviceName: {{ include "kafka.fullname" . }}-headless
   podManagementPolicy: {{ .Values.podManagementPolicy }}
+{{- if .Values.imagePullSecret }}
+  imagePullSecrets:
+    - name: {{ tpl .Values.imagePullSecret . }}
```

Comment:
why not just `{{ .Values.imagePullSecret }}`?

Reply:
Because it's pretty common to create the secret as part of the same helm chart you are deploying kafka in and then you are likely to make the secret `{{ .Release.Name }}-something` and you can't template inside the vaules.yaml which is where you'd set this. See:
https://github.com/helm/helm/issues/2492 and associated tickets.

Reply:
thanks @mcdan